cmake/winml.cmake

```diff
@@ -116,34 +117,99 @@ set_target_properties(winml_lib_telemetry
+   ${winml_lib_api_ort_dir}/pch.h
+  )
+
+# Compiler options
+target_compile_features(winml_lib_ort PRIVATE cxx_std_17)
+target_compile_options(winml_lib_ort PRIVATE /GR- /await /wd4238)
+
+# Compiler definitions
+target_compile_definitions(winml_lib_ort PRIVATE PLATFORM_WINDOWS)
+target_compile_definitions(winml_lib_ort PRIVATE _SCL_SECURE_NO_WARNINGS)                         # remove warnings about unchecked iterators
+
+# Specify the usage of a precompiled header
+target_precompiled_header(winml_lib_ort pch.h)
+
+# Includes
+target_include_directories(winml_lib_ort PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/winml_api)                   # windows machine learning generated component headers
+target_include_directories(winml_lib_ort PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/winml_api/comp_generated)    # windows machine learning generated component headers
+target_include_directories(winml_lib_ort PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/winml/sdk/cppwinrt/include)  # sdk cppwinrt headers
+
+target_include_directories(winml_lib_ort PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
+
+target_include_directories(winml_lib_ort PRIVATE ${REPO_ROOT}/winml)
+target_include_directories(winml_lib_ort PRIVATE ${winml_lib_api_dir})                            # needed for generated headers
+target_include_directories(winml_lib_ort PRIVATE ${winml_lib_api_core_dir})
+target_include_directories(winml_lib_ort PRIVATE ${winml_lib_api_ort_dir})
+target_include_directories(winml_lib_ort PRIVATE ${winml_lib_common_dir}/inc)
+target_include_directories(winml_lib_ort PRIVATE ${ONNXRUNTIME_INCLUDE_DIR})
+target_include_directories(winml_lib_ort PRIVATE ${ONNXRUNTIME_ROOT})
+
+set_target_properties(winml_lib_ort
+  PROPERTIES
+  FOLDER
+  ${target_folder})
+
+# Add deps
+add_dependencies(winml_lib_ort winml_sdk_cppwinrt)
+add_dependencies(winml_lib_ort winml_api)
+add_dependencies(winml_lib_ort winml_api_native)
+add_dependencies(winml_lib_ort winml_api_native_internal)
+
+# Link libraries
+target_link_libraries(winml_lib_ort PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/packages/DirectML.0.0.1/build/DirectML.targets)
+target_link_libraries(winml_lib_ort PRIVATE wil)
+
+
 ###########################
 # Add winml_adapter
 ###########################
 
 list(APPEND winml_adapter_files
-    ${winml_adapter_dir}/CpuOrtSessionBuilder.cpp
-    ${winml_adapter_dir}/CpuOrtSessionBuilder.h
+    ${winml_adapter_dir}/AbiCustomRegistryImpl.cpp
+    ${winml_adapter_dir}/AbiCustomRegistryImpl.h
     ${winml_adapter_dir}/CustomRegistryHelper.h
-    ${winml_adapter_dir}/FeatureDescriptorFactory.cpp
-    ${winml_adapter_dir}/FeatureDescriptorFactory.h
-    ${winml_adapter_dir}/LotusEnvironment.cpp
-    ${winml_adapter_dir}/LotusEnvironment.h
     ${winml_adapter_dir}/pch.h
-    ${winml_adapter_dir}/WinMLAdapter.cpp
-    ${winml_adapter_dir}/WinMLAdapter.h
     ${winml_adapter_dir}/ZeroCopyInputStreamWrapper.cpp
     ${winml_adapter_dir}/ZeroCopyInputStreamWrapper.h
+    ${winml_adapter_dir}/winml_adapter_apis.h
```

Comment:
>winml_adapter_apis [](start = 25, length = 18)

The naming of these files follow orts lower casing - should these be upper cased to follow winml file name conventions?